tpm2/policy_v3.go

```diff
@@ -148,10 +158,140 @@ type staticPolicyData_v3 struct {
 	RequireAuthValue       bool
 }
 
+// policyOrData_v3 is the version 3 on-disk representation of policyOrTree.
+// It is a flattened tree which suitable for serializing - the tree is just
+// a slice of nodes, with each node specifying an offset to its parent node.
+type policyOrData_v3 = policyOrData_v0
+
 // pcrPolicyData_v3 represents version 3 of the PCR policy metadata for
 // executing a policy session, and can be updated. It has the same format
 // as version 2.
-type pcrPolicyData_v3 = pcrPolicyData_v2
+type pcrPolicyData_v3 struct {
+	Selection                 tpm2.PCRSelectionList
```

Comment:
couldn't this just embed pcrPolicyData_v2 and replace below only the methods that have changes? it's very hard to spot what are difference between the methods are atm, are they mostly in executeRevocationCheck ?

Reply:
Yes, the changes are only in `executeRevocationCheck`. I'll try what you suggested.

Reply:
Ok, I've done this now, although it did require a bit more work in the unit tests.